cms-contentful/it/com/commercetools/sunrise/cms/contentful/ContentfulCmsServiceIT.java

```diff
@@ -52,65 +46,221 @@ public void whenCouldNotFetchEntry_thenThrowException() throws Exception {
+        ContentfulCmsService contentfulCmsService = ContentfulCmsService.of(spaceId(), token(), "finn", PAGE_TYPE_ID_FIELD_NAME);
+
+        Optional<CmsPage> content = contentfulCmsService.page("finn", emptyList()).toCompletableFuture().get(5, TimeUnit.SECONDS);
+
+        assertThat(content).isPresent();
+        Optional<String> field = content.get().field("mediaOneFile");
+        assertThat(field).isPresent();
+        assertThat(field.get()).isEqualTo("//images.contentful.com/l6chdlzlf8jn/2m1NzbeXYUksOGIwceEy0U/4e3cc53a96313a4bd822777af78a3b4d/some-5.jpg");
+    }
+
+    @Test
+    public void whenAskForContentInMediaArray_thenGetUrlElements() throws Exception {
+        ContentfulCmsService contentfulCmsService = ContentfulCmsService.of(spaceId(), token(), "finn", PAGE_TYPE_ID_FIELD_NAME);
+
+        Optional<CmsPage> content = contentfulCmsService.page("finn", emptyList()).toCompletableFuture().get(5, TimeUnit.SECONDS);
+
+        assertThat(content).isPresent();
+        Optional<String> field = content.get().field("mediaManyFiles[0]");
+        assertThat(field).isPresent();
+        assertThat(field.get()).isEqualTo("//images.contentful.com/l6chdlzlf8jn/6j9p38phC0oU0g42aqUSc4/2eb0c261bc13353ed867b13076af6b1f/logo.gif");
+
+        field = content.get().field("mediaManyFiles[1]");
+        assertThat(field).isPresent();
+        assertThat(field.get()).isEqualTo("//images.contentful.com/l6chdlzlf8jn/27BPx56xMcuGKe8uk8Auss/335581cd9daf3e9d0de254313e36d43b/some-5.jpg");
+    }
+
+    @Test
+    public void whenAskForContentInArrayOutsideTheScope_thenGet() throws Exception {
+        ContentfulCmsService contentfulCmsService = ContentfulCmsService.of(spaceId(), token(), "finn", PAGE_TYPE_ID_FIELD_NAME);
+        Optional<CmsPage> content = contentfulCmsService.page("finn", emptyList()).toCompletableFuture().get(5, TimeUnit.SECONDS);
+
+        assertThat(content).isPresent();
+        Optional<String> field = content.get().field("array[3].textArrayField");
+        assertThat(field).isNotPresent();
+    }
+
+    @Test
+    public void whenAskForContentWithGermanOrEmptyLocales_thenGetGermanTranslation() throws Exception {
+        ContentfulCmsService contentfulCmsService = ContentfulCmsService.of(spaceId(), token(), "finn", PAGE_TYPE_ID_FIELD_NAME);
+
+        Optional<CmsPage> content = contentfulCmsService.page("finn", SUPPORTED_LOCALES).toCompletableFuture().get(5, TimeUnit.SECONDS);
+
+        assertThat(content).isPresent();
+        Optional<String> field = content.get().field("array[3].textArrayField[1]");
+        assertThat(field).isPresent();
+        assertThat(field.get()).isEqualTo("zwei");
+
+        content = contentfulCmsService.page("finn", emptyList()).toCompletableFuture().get(5, TimeUnit.SECONDS);
+
+        assertThat(content).isPresent();
+        field = content.get().field("array[3].textArrayField[1]");
+        assertThat(field).isPresent();
+        assertThat(field.get()).isEqualTo("zwei");
```

Comment:
Just to clarify, in this project we are not trying to re-create something like JCR. We just want to support the *view* of content from a *headless* CMS, and if you think about it the possibilities are very few.

We are having Contentful as a good example of this group, but the management it does with default locales is very particular. Our interface should not be misguided by that.

I have created an issue for that: #29